src/editor/editor.cluster.tsx

```diff
@@ -0,0 +1,53 @@
+import {useCallback, useState} from 'react';
+
+import {Button, Modal} from 'antd';
+
+import {createUseComponentHook} from '@utils/hooks';
+
+import {didEditorContentChange} from './editor.instance';
+
+type WarnUnsavedChangesModalProps = {
+  open?: boolean;
+  onClose?: () => void;
+};
+
+const WarnUnsavedChangesModal = (props?: WarnUnsavedChangesModalProps) => {
+  const {open, onClose} = props || {};
+  return (
+    <Modal
+      open={open}
+      centered
+      footer={
+        <div>
+          <Button onClick={onClose}>Cancel</Button>
+          <Button>Discard changes</Button>
+          <Button>Deploy changes</Button>
+        </div>
+      }
+    >
+      <div>
+        <p>Pick what you would like to do with your changes.</p>
+        <p>If you deploy them, they will be applied to the cluster.</p>
```

Comment:
append " and replace any existing object with the same name" ? or is that too obvious?